src/executors.jl

```diff
@@ -355,8 +338,16 @@ function dispatch!(exec::Executor, graph::DispatchGraph; throw_error=true)
         (ExponentialBackOff(; n=retries(exec)), allow_retry(retry_on(exec)))
     end
 
-    f = wrap_on_error(
-        wrap_retry(
+    f1 = Dispatcher.wrap_on_error(
```

Comment:
Give `f1` and `f2` descriptive names like `wrapped_reset` and `wrapped_run` (or something else).